src/app/modules/main/chat_section/controller.nim

```diff
@@ -135,6 +135,12 @@ proc init*(self: Controller) =
     self.chatService.updateUnreadMessagesAndMentions(args.chatId, args.allMessagesMarked, args.messagesCount, args.messagesWithMentionsCount)
     self.delegate.onMarkAllMessagesRead(chat)
 
+  self.events.on(message_service.SIGNAL_MESSAGE_MARKED_AS_UNREAD) do(e:Args):
+    let args = message_service.MessageMarkMessageAsUnreadArgs(e)
+    let chat = self.chatService.getChatById(args.chatId)
+    self.chatService.updateUnreadMessage(args.chatId, args.messagesCount, args.messagesWithMentionsCount)
+    self.delegate.onMarkMessageAsUnread(chat)
```

Comment:
This looks a bit like a ping-pong in between layers: `messages service` -> `controller` -> `chat service`.

Ideally, it should be: 
`messages service` -> `chat service` then
`messages service` -> `controller`

In other words, if there is no listener to `SIGNAL_MESSAGE_MARKED_AS_UNREAD` then chat service wouldn't be updated, but it should.

Reply:
In short, our architecture looks as follows:
services | controller - module - view

**services**  is the source of truth and is the place where we communicate with status-go.
**controller** is the only one who talk with services.
**module** is core logic for given module, uses controller to get/set the data, knows anything about services.
**view** is QtObject that is exposed to qml and talks only with module.

Services can talk to each other, directly. Messages service has already reference to chat service, to update the chat when it is needed, see: https://github.com/status-im/status-desktop/blob/ab7085a18424062dc0dc63c8fe55ec6d74dec7f1/src/app_service/service/message/service.nim#L293

I will try to elaborate more about architecture on Monday.

Reply:
Got it, and you're right that service can talk to each others, I just thought in my mind the ask was to capture the signal  `SIGNAL_MESSAGE_MARKED_AS_UNREAD` inside the chat_service. That's the main reason why I expressed my concern.
I think I can submit another code change.
Thanks a lot @osmaczko !

Reply:
![StatusAppHLClassDiagram](https://github.com/status-im/status-desktop/assets/33099791/7ace8d64-3554-45b7-b54d-39cc7da11bef)

That explains the architecture we try to follow pretty well. The diagram itself was done by @saledjenic some time ago already. FYI: recently there was a discussion about a bit different architecture, you can follow it there https://github.com/status-im/status-desktop/pull/12489

Reply:
Thanks for the share @osmaczko, this is gold 💰